coding-intro/coding-intro-core/core-concepts/programs-and-data.md

```diff
@@ -8,35 +8,77 @@ category: must-know
 links:
   - >-
     [Exact Instructions Challenge](https://youtube.com/watch?v=Ct-lOOUqmyY){video}
-  
+practiceQuestion:
+  formats:
+    - fill-in-the-gap
+  context: standalone
+
 ---
 
 # Programs & Data
 
 ---
 ## Content
 
-Let's quickly introduce the building blocks we'll cover in this course.
+Let's introduce the building blocks we'll cover in this course.
+
+
+### Programs
+
+Today you might have woken up with a to-do list, which is essentially your program for the day.
+
+Similarly, we give the computers a to-do list of extremely specific commands, together called a program, to carry out.
+
+
+> 💡 Scroll down to the Learn more section to see a hilarious example of how specific instructions computers need 😆
+
+
+### Data
 
-### Program
+Data is the information that our programs work with. In code, we store these in variables.
 
-Today you woke up with a plan, a to-do list, which is essentially your program for the day. Similarly, we give the computers a set of commands to carry out, one after the other. That too is known as a program.
+For instance, we could store someone's `name` and `age` in variables[1]:
+```python
+name = "Batman"
+age = 30
+```
 
-The only difference being that computers require specific instructions. (Don't tell anyone we said this, but machines are actually pretty dumb… sssh!)
 
-> 💡 Scroll down to see a hilarious real life example of this in the Learn more section 😆
+### Types
 
+Now remember, computers are extremely literal machines! If you write a program to divide "Batman" by 2, it is going to try (and fail) to do that.
 
-### Data: Variables
+That's why we group our data into distinct **data types**, the three main ones being:
+
+| Type    | Examples    |
+| ------- | ----------- |
+| String  | "Batman"    |
+| Number  | 5,-0.5     |
+| Boolean | True, False |
+
+What would we get if we tried to take the square root of `"cat"`?
```

Comment:
We never explained what happens when you perform operations on a type that aren't appropriate for that type. It might be confusing to beginners how to answer this.

Before we explained:

> These data types save us from attempting number related operations on texts and vice versa.
>
> For instance, what's the square root of `"cat"`?

Note: Since this insight ends in a question, lets also change the category of this insight into "discussion".